pkg/controller/oneagent/oneagent_update.go

```diff
@@ -174,14 +175,39 @@ func (r *ReconcileOneAgent) findOutdatedPodsImmutableImage(logger logr.Logger, i
 	return outdatedPods, err
 }
 
-func isLatest(logger logr.Logger, image string, imageID string, imagePullSecret *corev1.Secret) (bool, error) {
-	dockerConfig, err := utils.NewDockerConfig(imagePullSecret)
-	if err != nil {
-		logger.Info(err.Error())
-	}
+func isLatest() func(logr.Logger, string, string, *corev1.Secret) (bool, error) {
+	failed := false
+	cache := map[string]bool{}
+
+	return func(logger logr.Logger, image string, imageID string, imagePullSecret *corev1.Secret) (bool, error) {
+		if failed {
+			return true, nil
+		}
+
+		if latest, ok := cache[imageID]; ok {
+			return latest, nil
+		}
+
+		logger.Info("Fetching image hash")
 
-	dockerVersionChecker := utils.NewDockerVersionChecker(image, imageID, dockerConfig)
-	return dockerVersionChecker.IsLatest()
+		dockerConfig, err := utils.NewDockerConfig(imagePullSecret)
+		if err != nil {
+			failed = true
+			logger.Info(err.Error())
+			return true, nil
+		}
+
+		dockerVersionChecker := utils.NewDockerVersionChecker(image, imageID, dockerConfig)
+		latest, err := dockerVersionChecker.IsLatest()
+		if err != nil {
+			failed = true
+			logger.Info(err.Error())
+			return true, nil
+		}
+
+		cache[imageID] = latest
```

Comment:
is there a case where imageID can be nil?

Reply:
It's checked beforehand: https://github.com/Dynatrace/dynatrace-oneagent-operator/blob/78481d65deb6d69843f3ca00825a6c2fa58b2aaf/pkg/controller/oneagent/oneagent_update.go#L144-L161